ncm-ceph/src/main/pan/components/ceph/schema.pan

```diff
@@ -187,19 +187,39 @@ type ceph_cluster_config = {
     'fsid' : type_uuid # Should be generated with uuidgen
     'mon_cluster_log_to_syslog' : boolean = true
     'mon_initial_members' : type_network_name [1..]
+    'mon_osd_down_out_interval' ? long(0..)
+    'mon_osd_down_out_subtree_limit' ? string # takes values of bucket types in crushmap, some of which are user definable
     'mon_osd_min_down_reporters' ? long(0..)
     'mon_osd_min_down_reports' ? long(0..)
     'mon_osd_max_op_age' ? long = 32
     'ms_type' ? string with match(SELF, '^(simple|async|xio)$')
+    'osd_backfill_scan_max' ? long(0..)
+    'osd_backfill_scan_min' ? long(0..)
+    'osd_client_op_priority' ? long(1..63)
+    'osd_crush_chooseleaf_type' ? boolean
     # the component sets this to false when generating crushmap itself, and true when crushmap is generated by ceph
     'osd_crush_update_on_start' ? boolean
+    'osd_disk_thread_ioprio_class' ? string with match(SELF, '^(idle|be|rt)$')
+    'osd_disk_thread_ioprio_priority' ? long(0..)
+    'osd_disk_threads' ? long(0..)
+    'osd_heartbeat_grace' ? long(0..)
+    'osd_heartbeat_interval' ? long(0..)
     'osd_journal_size' : long(0..) = 10240
+    'osd_max_backfills' ? long(0..)
+    'osd_max_scrubs' ? long(0..)
```

Comment:
Some of these new options are already defined at https://github.com/quattor/configuration-modules-core/blob/master/ncm-ceph/src/main/pan/components/ceph/schema-osd.pan. I guess we shouldn't add non-global options here, but in the actual schema-<daemon> template. If you would need them as global config options, we can include the options here alltogether.

Reply:
This comment is still valid :smile:

Reply:
What do you propose we do with these?